File: 4. Recursion/str_length.py
str = "GeeksforGeeks"
 
# Function to
# calculate length
def string_length(str) :
     
    # if we reach at the
    # end of the string
    if str == '':
        return 0
    else :
        return 1 + string_length(str[1:])
     
# Driver Code
print (string_length(str))

File: 4. Recursion/str_length.js
/* Function to calculate length */
function recLen(str) {

    // if we reach at the end of the string
    if (str == "")
        return 0;
    else
        return recLen(str.substring(1)) + 1;
}

// Driver code

let str = "GeeksforGeeks";
console.log(recLen(str));